src/cobra/core/object.py

```diff
@@ -1,31 +1,53 @@
-# -*- coding: utf-8 -*-
+"""Define base Object class in Cobra."""
 
-from __future__ import absolute_import
+from typing import Optional
 
 
-class Object(object):
-    """Defines common behavior of object in cobra.core"""
+class Object:
+    """Defines common behavior of object in cobra.core."""
 
-    def __init__(self, id=None, name=""):
-        """A simple object with an identifier
+    def __init__(self, id: Optional[str] = None, name: str = "") -> None:
+        """Initialize a simple object with an identifier.
 
         Parameters
         ----------
-        id: None or a string
+        id: string, optional
             the identifier to associate with the object
+        name: string, optional
+            The name to associate with the object. Default "".
+
+        Objects will have notes and _annotation as dicitionaries, initialized as empty
+        dictionaries.
         """
         self._id = id
         self.name = name
 
-        self.notes = {}
-        self._annotation = {}
+        self.notes = dict()
+        self._annotation = dict()
```

Comment:
Prefer the literal instance when possible.

```suggestion
        self.notes = {}
        self._annotation = {}
```